src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py

```diff
@@ -0,0 +1,431 @@
+        seed=42,
+        num_inference_steps=50,
+        neg_prompt="",
+        prompt_strength=1.0,
+        prompt_reps=20,
+        use_ddim=False,
+    ):
+
+        cond_image = samples["cond_images"]  # reference image
+        cond_subject = samples["cond_subject"]  # source subject category
+        tgt_subject = samples["tgt_subject"]  # target subject category
+        prompt = samples["prompt"]
+        cldm_cond_image = samples.get("cldm_cond_image", None)  # conditional image
+
+        prompt = self._build_prompt(
+            prompts=prompt,
+            tgt_subjects=tgt_subject,
+            prompt_strength=prompt_strength,
+            prompt_reps=prompt_reps,
+        )
+
+        text_embeddings = self._forward_prompt_embeddings(
+            cond_image, cond_subject, prompt
+        )
+        # 3. unconditional embedding
+        do_classifier_free_guidance = guidance_scale > 1.0
+        if do_classifier_free_guidance:
+            max_length = self.text_encoder.text_model.config.max_position_embeddings
+
+            uncond_input = self.tokenizer(
+                [neg_prompt],
+                padding="max_length",
+                max_length=max_length,
+                return_tensors="pt",
+            )
+            uncond_embeddings = self.text_encoder(
+                input_ids=uncond_input.input_ids.to(self.device),
+                ctx_embeddings=None,
+            )[0]
+            # For classifier free guidance, we need to do two forward passes.
+            # Here we concatenate the unconditional and text embeddings into a single batch
+            # to avoid doing two forward passes
+            text_embeddings = torch.cat([uncond_embeddings, text_embeddings])
+
+        if seed is not None:
+            generator = torch.Generator(device=self.device)
+            generator = generator.manual_seed(seed)
+
+        latents = self._init_latent(latents, height, width, generator, batch_size=1)
+
+
+        # set timesteps
+        extra_set_kwargs = {}
+        self.scheduler.set_timesteps(num_inference_steps, **extra_set_kwargs)
+
+        iterator = tqdm.tqdm(self.scheduler.timesteps)
+
+        for i, t in enumerate(iterator):
+            latents = self._denoise_latent_step(
+                latents=latents,
+                t=t,
+                text_embeddings=text_embeddings,
+                cond_image=cldm_cond_image,
+                height=height,
+                width=width,
+                guidance_scale=guidance_scale,
+                use_inversion=use_ddim,
+            )
+
+        image = self._latent_to_image(latents)
+        
+
+
+        return image
+
+    def _latent_to_image(self, latents):
+        latents = 1 / 0.18215 * latents
+        image = self.vae.decode(latents).sample
+        print(torch.mean(image))
+        image = (image / 2 + 0.5).clamp(0, 1)
+        image = image.cpu().permute(0, 2, 3, 1).numpy()
+
+        image = numpy_to_pil(image)
+
+        return image
```

Comment:
We should make use of https://github.com/huggingface/diffusers/blob/078df46bc9a99178a9a744b872899990353769a4/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py#L398